staging/src/k8s.io/component-base/logs/datapol/externaltypes.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package datapol
+
+import (
+	"net/http"
+	"reflect"
+
+	"golang.org/x/oauth2"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+// GlobalDatapolicyMapping returns the list of sensitive datatypes are embedded
+// in types not native to Kubernetes.
+func GlobalDatapolicyMapping(v interface{}) []string {
+	return byType(reflect.TypeOf(v))
+}
+
+func byType(t reflect.Type) []string {
+	switch t {
+	case reflect.TypeOf(http.Header{}):
+		return []string{"password", "token"}
+	case reflect.TypeOf(http.Cookie{}):
+		return []string{"token"}
+	case reflect.TypeOf(oauth2.Token{}):
```

Comment:
Checking this way for type means that `component-base` will take on lot's of dependencies that are not related to it's functionallity. Is there a way that we could avoid this?

Reply:
I changed the type checking to be purely on the strings, the dependencies are test only now, which would hopefully alleviate some of the concerns.

Reply:
Hmm, working on strings feels fragile, since when building in bazel the resulting types are different than when i just did a `go test` on my local machine